query-translator/src/translate.rs

```diff
@@ -148,14 +154,69 @@ pub struct ProjectedSelect{
     pub projector: Box<Projector>,
 }
 
+// Nasty little hack to let us move out of indexed context.
+struct ConsumableVec<T> {
+    inner: Vec<Option<T>>,
+}
+
+impl<T> ConsumableVec<T> {
+    fn with_vec(vec: Vec<T>) -> ConsumableVec<T> {
+        ConsumableVec { inner: vec.into_iter().map(|x| Some(x)).collect() }
+    }
+
+    fn take_dangerously(&mut self, i: usize) -> T {
+        ::std::mem::replace(&mut self.inner[i], None).expect("each value to only be fetched once")
+    }
+}
+
+fn table_for_computed(computed: ComputedTable, alias: TableAlias) -> TableOrSubquery {
+    match computed {
+        ComputedTable::Union {
+            projection, type_extraction, arms,
+        } => {
+            // The projection list for each CC must have the same shape and the same names.
+            // The values we project might be fixed or they might be columns, and of course
+            // each arm will have different columns.
+            // TODO: type extraction.
+            let queries = arms.into_iter()
+                              .map(|cc| {
+                                    let var_columns = projection.iter().map(|var| {
+                                        let col = cc.column_bindings.get(&var).unwrap()[0].clone();
+                                        ProjectedColumn(ColumnOrExpression::Column(col), var.to_string())
+                                    }).collect();
+                                    let projection = Projection::Columns(var_columns);
+                                    cc_to_select_query(projection, cc, false, None)
```

Comment:
They're used in the next commit; see the commit message that explains we haven't got to that yet.